src/components/shared/hooks.js

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react'
+
+// If the input has a value (checkedValue prop) the form value is: checkedValue || ''
+// Otherwise the form-value is: true || false
+const getToggleValue = ({ checked, value }) => {
+    if (value) {
+        return checked ? value : ''
+    }
+    return checked
+}
+
+const useToggleChangeHandler = onChange =>
+    useCallback(payload => {
```

Comment:
I would refrain from using `useCallback` here. This article makes the point better than I can: https://kentcdodds.com/blog/usememo-and-usecallback

To summarize it, the computation here is not expensive, and there isn't anything this seems to be solving other than performance optimization (unless I'm missing something), so we're better off keeping the complexity down. It might even hurt performance to use useCallback here.

Reply:
Good point @ismay. I am currently working on rewriting the `FieldAdapter` component as a hook and can definitely relate to the "difficult to comprehend" part, as I was trying to theorise about which dependencies I'd need and when these would trigger a rerender......

TBH... I haven't worked with hooks much and was simply following the pattern that you seem to see everywhere on the web: need an eventHandler? useCallBack! It's very useful you've pulled me up on simply copying that pattern. 

I've read the article you've linked to and while I agree with the point Kent C. Dodds is trying to make, I think his examples in most of the article are giving the hooks an unfair disadvantage because:
- Focus seems to be fully on the first render, and obviously memoized stuff only starts to shine when being called repetitively and with the same arguments...
- And that also brings me to the next point, in the vast majority of examples in the article, the dependency array is empty which basically means that no memoization is happening at all.

Anyway... 
Since no performance issues have been identified, there is nothing to optimise and I would be in favour of removing `useCallback` and `useMemo` for simplicity's sake. But, on the other hand, I could also be swayed to keep them in, if the current implementation is idiomatic hooks style. Unfortunately, I don't feel I've worked with hooks enough yet to form a solid opinion on this.

@Mohammer5 @varl please share your views on this too